airflow/jobs/scheduler_job.py

```diff
@@ -403,6 +403,17 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                     # Many dags don't have a task_concurrency, so where we can avoid loading the full
                     # serialized DAG the better.
                     serialized_dag = self.dagbag.get_dag(dag_id, session=session)
+                    # If the dag is missing, fail the task and continue to the next task.
+                    if not serialized_dag:
+                        self.log.error(
+                            "DAG '%s' for task instance %s not found in serialized_dag table",
+                            dag_id,
+                            task_instance,
+                        )
+                        session.query(TI).filter(TI.dag_id == dag_id, TI.state == State.SCHEDULED).update(
+                            {TI.state: State.NONE}, synchronize_session='fetch'
```

Comment:
I don't think we should set it to `None` ! We should Mark it failed for sure. There is also `TaskInstanceState.REMOVED` which partially applies and partially doesn't as it is not the task alone that is removed but the entire dag